strawberryfields/apps/graph/resize.py

```diff
@@ -245,3 +236,28 @@ def greedy_degree(
 """dict[str, func]: Included methods for resizing subgraphs. The dictionary keys are strings
 describing the method, while the dictionary values are callable functions corresponding to the
 method."""
+
+
+def clique_shrink(subgraph: list, graph: nx.Graph) -> list:
+    """Shrinks an input subgraph until it forms a clique.
+
+    Args:
+        subgraph (list[int]): a subgraph specified by a list of nodes
+        graph (nx.Graph): the input graph
+
+    Returns:
+        list[int]: a clique of size smaller than or equal to the input subgraph
+    """
+
+    if not utils.is_subgraph(subgraph, graph):
+        raise ValueError("Input is not a valid subgraph")
+
+    subgraph = graph.subgraph(subgraph).copy()  # Create NetworkX Graph version of subgraph
```

Comment:
This strikes me as an unnecessary comment. I don't have much experience with networkX but it replicates a common pattern in "plain" python (`copy` exists on lists, dicts, etc., and performs roughly the same function) that I understood it anyway. 

FWIW, when you call `graph.subgraph`, NetworkX returns a subgraph that is really just a "view" into the existing graph, without actually storing a new data structure in memory (similar to nested dictionary access/assignment). Apparently NetworkX has explicitly frozen this structure to prevent accidental modification, which is a bit strange in my opinion, but you would have needed to do this anyway to prevent modification to the original graph object.